### DISCUSSION TITLE
How to get started with Wican for my Opel Ampere-E

### DISCUSSION THREAD

**Discussion by brjhaverkamp** [meatpiHQ/wican-fw#237] [General]
Hi all,

I’ve been starting with the Wican and my Opel Ampera-e/GM Chevrolet Bolt 2017
Sofar I have been able to get the HA integration up and running and upgraded the Wican firmware from 2.98 to 3.48. So I can start with the Automate tab in the web interface.

I found the list of available PIDs on [Chevy Bolt OBD2 PIDs – All EV Info(https://allev.info/boltpids/)
And I have been trying to implement them in the Wican Automate tab. This is what I did so far:
![Screenshot from 2024-10-16 19-46-30](https://github.com/user-attachments/assets/5f215c7c-bb92-423b-b8d2-955ca83cedb2)
a/82a62ad1c901d2cef680fa016e023aeb6eca513c.png)

From what I understood, int the expressions in the file from boltpids A, B, C, etc. are byte 0, 1, 2. in that files syntax.
In the Wican syntax that means A=B0, B=B1, C=B2, etc. Is that correct?
I just tried two from the csv list that looked interesting. But I have no idea what the Custom Initialization string is. (Is that always needed?) and the Init part of the Custom PIDs. What should I fill in there?

And I assume I can leave the “Vehicle Specific” at the top of the automate tab to “Disabled”, right?

I do see mqtt messages from the Wican in Node-Red. But mostly about getting online and offline. No data from these custom PIDs yet.

Any pointers on the above items, or other advice is welcome.

Bert

P.S. I posted this earlier in the Home assistant forum, but I was, rightly so, directed here

**Comment by brjhaverkamp** [comment]
Quick update, 
As I couldn't get any data from the Wican, I spent a while searching for Chevy Bolt or Opel Ampera-E related information on the custom initialization strings. But I didn't find anything. Is that init string not important?

Then I tried switching on the car an see if that would give me more data. 
Indeed, i get a huge flow of messages from Wican, but I can't make sense of it. 
In node-red I monitor for all messages from Wican/#. I now get this kind of messages. Can anyone tell me what do they mean?
![Screenshot from 2024-10-20 21-19-09](https://github.com/user-attachments/assets/f5fe61c9-e741-409b-9ae1-723738867ad8)

And I was hoping to get data on the battery status when my car is parked. Is that possible, or does the contact have to be switched on? 

Regards,

Bert

**Reply by meatpiHQ** [reply]
@brjhaverkamp I explain it in the link below.

> But I didn't find anything. Is that init string not important?

Yes the init string is important. Please export the logs from car scanner app and I can help you create a profile for you car.

<img width="398" alt="image" src="https://github.com/user-attachments/assets/499e71dc-5226-481f-80da-7e4da54ffaa7">


https://github.com/meatpiHQ/wican-fw/issues/165#issuecomment-2275560958

**Reply by brjhaverkamp** [reply]
Hi,

That would be much appreciated. 

I even found the car scanner app your screenshot
 I choose the first one that came up: "Car scanner ELM ODB2". Sounded appropriate, and the color scheme matched your screenshot:-)  It took me a while to connect. Found out the hard way that the car has to be turned on. (Ignition on doesn't sound right in an EV...)

And behold, after a bit of browsing through the app I found the page that matched your screenshot!
Here is the logfile. 

[Opel Ampera-E car scanner log.txt](https://github.com/user-attachments/files/17465396/Opel.Ampera-E.car.scanner.log.txt)

**Reply by brjhaverkamp** [reply]
One more question: (bit of expectation management towards myself:-) I was hoping I could read the battery status in Home assistant when I have the car parked next to the house.
Will that work? Or will I only be able to read it out when the car is still switched on? And how about when it's on the charger?

Regards, Bert

**Reply by meatpiHQ** [reply]
@brjhaverkamp I've created a new profile for the Ampere-E, you should disable mqtt and follow the documentation 

https://meatpihq.github.io/wican-fw/home-assistant/integration_setup

> Will that work? Or will I only be able to read it out when the car is still switched on? And how about when it's on the charger?

This depends on the car some cars work while charging, some need to be charging and unlocked. The only way to know is to test.

**Comment by brjhaverkamp** [comment]
Thanks for the profile. I just added it. This works like a charm! Very nice.
I did assume I have to re-enable mqtt afterwards:-)
I also set the mqtt topic to wican/amperae/pids and the cycle time to 5000ms.

I see  two things that worry me.  Firstly, when in Node-Red I monitor wican/#   (# is the wildcard in Node-Red) there is an enormous flood of messages on wican/amperae/rx (the RX topic I set in the settings tab) Litterally 100s per second
The format of those messages is like I shared in the screenshot 2 days ago.

Secondly, when I monitor on wican/amperae/pids, I get these messages. 
![Screenshot from 2024-10-22 19-51-45](https://github.com/user-attachments/assets/b2d8f811-a851-43cf-b945-60d25466c2cd)
So it looks like something is still not right in the profile:-(

Some additional question: 
- I saw in the profile that you set the init to ATSP6;ATST96. How did you arrive at that? I checked the log for these strings, but couldn't find anything special. It looks like magic to me:-)
- What are the rx, tx and status topics for that you can specify in the settings tab? What messages should I see there normally?
-  Small typo: Can you change ampere to ampera?   (an e->a, both in the filename and within the file.
- Is it possible to download the profile from the wican itself? So that I can change and expand it and upload it again to try. Now I had to go to github to find the ampere-e.json.
-  And finally as a feature request, is it possible to download the custom-pids in the correct Json format, so you can test them as custom and then add them to a profile?

Regards,

Bert

**Reply by meatpiHQ** [reply]
> I did assume I have to re-enable mqtt afterwards

No if you're using the HA integration you should keep mqtt disabled. The profile should work properly after you disable mqtt.

> ATSP6;ATST96. How did you arrive at that?

From car scanner log, this is just an initialization string

> rx, tx and status topics for that you can specify in the settings tab?

You should look at the documentation. The messages you're receiving are the messages broadcasted on the bus.

https://meatpihq.github.io/wican-fw/mqtt/usage

> Is it possible to download the profile from the wican itself? So that I can change and expand it and upload it again to try. Now I had to go to github to find the ampere-e.json.

Here is the [profiles file](https://github.com/meatpiHQ/wican-fw/blob/main/vehicle_profiles.json) you can remove other Vehicles and upload it in the Automate page

> And finally as a feature request, is it possible to download the custom-pids in the correct Json format, so you can test them as custom and then add them to a profile?

This actually a good idea!

**Reply by brjhaverkamp** [reply]
Hi,

Thanks for the answers. So what I saw was all the CAN bus messages, passed on over MQTT. 
What I was looking for is simultaneously get the PIDs via MQTT into Node-Red and into HA via the integration for showing them in a dashboard. Is that possible? 
But I don't need all the CAN bus message, as that only adds extra load on my mqtt server.  I noticed that there is a CAN mode  Normal/Silent. Would setting that to silent work?

I tried it, and it seems to work:-)

So I am making progress. What I can't get is any PIDs unfortunately. With the Opel Ampera-E profile, in HA, I get SOC: unavailable
![Screenshot from 2024-10-26 10-01-11](https://github.com/user-attachments/assets/ec2cf5ee-1708-43ce-b144-1509d9a9184c)
In Node-RED, I get a bit more info. It reports a time-out error. 
![Screenshot from 2024-10-26 10-02-41](https://github.com/user-attachments/assets/1fd90a04-6130-4326-a2f6-3f6ffbfb8c7c)

I'll keep on digging to see if I can find more clues.

Regards, Bert

**Reply by meatpiHQ** [reply]
@brjhaverkamp 

> With the Opel Ampera-E profile, in HA, I get SOC: unavailable

I've just updated the profile, can you please sync the profile on your device, then click store and give it a reboot. And if you still don't see the SoC please disable the MQTT and try again.

<img width="445" alt="image" src="https://github.com/user-attachments/assets/19c2087f-cabb-43ed-a355-fd55c1c2856e">

**Reply by brjhaverkamp** [reply]
Hi meatpiHQ, 

I synced. Checked in github also on what has changed. You changed the name of the second PID from SOC to Capacity, right?
Unfortunately, I still don't see any change. Neither with mqtt enabled and without. I do see other responses in the HA intergration. E.g. changing the CAN mode, or the ECu mode switching on and off is nicely reflected in the Logbook of the Wican integration. So the basic communication is working.

One thing I noticed is that you appended a "1" to the end of both PIDs. So 228334 ->2283341, and 2241A3->2241A31 Is the "1" signifying something?

I tried to edit the json file locally and remove the "1". Then I uploaded it via the browse option in the Automate tab. But also that didn't work. I tried some more variants of the init string. But again no luck.

I'll keep on digging

Bert

**Reply by meatpiHQ** [reply]
@brjhaverkamp Sorry for the delay! Can you please post a screenshot of your device configuration page? Both Setting and Automate tabs. 

> One thing I noticed is that you appended a "1" to the end of both PIDs. So 228334 ->2283341, and 2241A3->2241A31 Is the "1" signifying something?

The '1' means it should expect one frame response.

**Reply by SAUhrich** [reply]
Hi @meatpiHQ and @brjhaverkamp ,
I will not hijack this thread, but just letting you know that I am following to get tips on configuring a similar setup for my 2023 Chevy Bolt. I just last week received my WiCAN-ODB and have only gotten as far as connecting it to the local lan, setup and configured with Car Scanner and captured the ELM logs. I've also got the entities created in Home Assistant. 

I will be working over the next couple of weeks when work permits on trying to decipher the correct PIDs for trying to get my SoC to show up properly in Home Assistant. I think Bert, I have the exact same goals: I need to monitor the SoC in Home Assistant to trigger the correct timing for initiating the charge cycle from HA based on other factors (home solar/batteries config'd in HA as well).

Although the year difference in our vehicles could be a factor in the PID discovery etc, I would assume that they would be relatively similar, I will share anything I find useful here in this thread. Otherwise I will not cloud up yall's discussion and progress, and just lurk to get info :) And thanks to both of your dialogs for getting me this far!

**Reply by meatpiHQ** [reply]
@SAUhrich I think it would be better to open a new issue for your Chevy Bolt. It would be easier for user with similar car to find it. You can post your logs in the new issue and I'll help you create a profile.

**Reply by SAUhrich** [reply]
Will do this evening, and thank you.

**Reply by brjhaverkamp** [reply]
> @brjhaverkamp Sorry for the delay! Can you please post a screenshot of your device configuration page? Both Setting and Automate tabs.
> 
> > One thing I noticed is that you appended a "1" to the end of both PIDs. So 228334 ->2283341, and 2241A3->2241A31 Is the "1" signifying something?
> 
> The '1' means it should expect one frame response.

Ah, almost missed your post due to Suhrich's reply, just after it. Thanks for explaining the 1 in the PID.

Attached are the 2 screenshots. Did I missconfigure anything?
![Screenshot from 2024-11-11 20-27-45](https://github.com/user-attachments/assets/a449a8cd-99cc-4e36-869a-6b5ecaec12c3)
![Screenshot from 2024-11-11 20-27-52](https://github.com/user-attachments/assets/a3b8062d-2063-46e0-956a-df84682207de)

**Comment by brjhaverkamp** [comment]
Hi SAUhrich. Good to hear  you are  trying to get the Wican working as well on a Chevy Bolt/Opel Ampera-E  I hope we can work together. To be honest I am stuck a bit, I still haven't been able to get _any_ data out. It looks like I am stuck with the initialization. 
With the car scanner app I was able to get data on my phone. But I haven't had any  luck yet via MQTT or the Home assistant integration. 

Once we have passed that hurdle, I hope it is smooth sailing. There is a long list of PIDs and their expresssions on https://allev.info/boltpids/ 
I think you are right and the model make year does'n't matter too much. So if you make the issue, feel free to include the Ampera-E and we can work on it together. And I hope meatpiHQ can make a profile for you:-)

Bert

**Reply by SAUhrich** [reply]
Hi Bert, I think a recent update to the vehicle profile must have been what gave me success, I posted my results in the Bolt thread I started here: #256 . This is not using MQTT, but the auto_pid method. I was particularly glad (and surprised!) that the ECU stays active when the car is off and during charging, so as long as the car is at home on WiFi, I get up to date SoC values.

I hope this is helpful to you, Thanks, I think go to @meatpiHQ for the profile!

**Reply by meatpiHQ** [reply]
@brjhaverkamp Please try to sync the profile again.

**Comment by dutchsteven** [comment]
I've managed to get my SOC into Home Assistant for my Opel Ampera-E, I guess it will also work for Bolt EV.
I'm not using the Automate-tab, but the old-school MQTT method

**Wican**
![image](https://github.com/user-attachments/assets/b5c059a6-7763-4419-9d75-ec3832cdb92f)
![image](https://github.com/user-attachments/assets/585cf2fe-e0e4-4fa5-964b-3b47b937ba92)

**Home Assistant** 
added to configuration.yaml
```
mqtt:
  sensor:
    - name: "Ampera-E SOC"
      unique_id: "amperae-soc"
      state_topic: "amperae/rx"
      unit_of_measurement: "%"
      state_class: "measurement"
      value_template: "{{ value_json.SOC | float(0) }}"
```

Add an automation

```
alias: "State of Charge Opel Ampera-e "
description: ""
mode: single
triggers:
  - minutes: /15
    trigger: time_pattern
conditions: []
actions:
  - metadata: {}
    data:
      qos: "0"
      topic: amperae/tx
      payload: >-
        {"bus":"0","type":"tx","ts":18174,"frame":[{"id":2020,"dlc":8,"rtr":false,"extd":false,"data":[3,34,131,52,170,170,170,170]}]}
    enabled: true
    action: mqtt.publish
```

The part in configuration.yaml creates the sensor 'sensor.ampera_e_soc' and fills it when the MQTT-topic 'amperae/rx' has been populated with the SOC.
The automation sends a command to topic 'amperae/tx', which in return gives a response on the 'amperae/rx' topic. 

This works best when the car is on or charging. So it happens that you first have to charge a few moments, run the automation, disable the charging. This way you get the SOC populated.

**Comment by brjhaverkamp** [comment]
Hi all, good news from the Netherlands!
First Dutchsteven (at least I asume you are from the Netherlands) reported a working Opel Ampera, And he even mentioned that the PIDs could be read when the car is charging. That was something I was not sure about yet.
 
And on my side, with the latest update from meatpiHQ, I got data! So I guess the PID init was the deciding change.
I now have a SOC and Capacity reading. the SOC is 25.88 and rising. Which is about equal to what I see on the graph in the car. In the car it is actually quite a bit above the 1/4th marker, but that might be a wrong visual..
The Capacity states 184.3 I have no idea what that means. It is definitely not my battery capacity in kWh.. I wish:-)

@meatpiHQ, am I correct that if I download the ampere-e.json and start adding or changing PIDs, and then upload it through the Browse button on the Automate tab, I can easily extend and test adding more PIDs to this profile? Although the SOC is by far the most important one for me.

Oh, and can you pleaserename the ampere-e to ampera-e  (E->A). You took over my typo from the title of this discussion. Appologies, but it is really an ampera-e..

Regards,

Bert

**Reply by meatpiHQ** [reply]
@brjhaverkamp  yes, but the format of the json you should upload is different.

I've created this file for you and  removed the other cars, so you can download this file and make your changes to it and upload it directly to the device [vehicle_profiles.json](https://github.com/user-attachments/files/17749033/vehicle_profiles.json)

**Reply by brjhaverkamp** [reply]
Hi  meatpiHQ. 

This vehicle_profiles.json is indeed slightly differently formatted than the ampere-e.json I downloaded from the vehicle_profiles/opel folder on your github. 
However, I still don't seem to get any changes accepted by the Wican.
I only changed 3 characters (start simple:-) I changed the name Capacity to Capacity_Ah.
Then I uploaded the this vehicle_profiles.json with the browse button in the automate tab
The interface then flips to the status tab. 
I went back to the Automate tab and pressed "store".
"Submit changes" is greyed out, so to be sure I rebooted via the button on the About tab.
But the new name doesn't show up in the HA integration. What should I do differently?

![Screenshot from 2024-11-15 08-43-54](https://github.com/user-attachments/assets/78e78e56-e0bf-4eb2-b246-eb4c17c33b70)

**Reply by meatpiHQ** [reply]
It seems that the vehicle specific is disabled? It should be enabled?

**Comment by brjhaverkamp** [comment]
Ah, I disabled it because it then asks to fill in a vehicle model. Which I don't want, because I want to use my own profile.
But I enabled it now, and no difference. Capacity is still called Capacity, instead of Capacity_Ah.
Is there a limitation on the characters I can use in a name?
Do I need to do something in the integration in HA to get the PIDs updated (Sofar they popped up automatically)

![Screenshot from 2024-11-15 10-50-22](https://github.com/user-attachments/assets/5343851f-5bb9-452e-a99b-7606bf350522)

**Reply by sparkelende** [reply]
> Ah, I disabled it because it then asks to fill in a vehicle model. Which I don't want, because I want to use my own profile. But I enabled it now, and no difference. Capacity is still called Capacity, instead of Capacity_Ah. Is there a limitation on the characters I can use in a name? Do I need to do something in the integration in HA to get the PIDs updated (Sofar they popped up automatically)
> 
> ![Screenshot from 2024-11-15 10-50-22](https://private-user-images.githubusercontent.com/3051118/386548298-5343851f-5bb9-452e-a99b-7606bf350522.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzI3NDY5MjksIm5iZiI6MTc3Mjc0NjYyOSwicGF0aCI6Ii8zMDUxMTE4LzM4NjU0ODI5OC01MzQzODUxZi01YmI5LTQ1MmUtYTk5Yi03NjA2YmYzNTA1MjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDMwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAzMDVUMjEzNzA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2RiN2Y4OGQ0ZDQ2MWJmZmFjOGNkZjNhNDVlMzBiNmZkM2Q5ODk1YTI2NDU5ZjQ5NDRmZDY3MjM1NjZlMTYxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.jSY1mNtRbbCyLSlgrZx-W6AUhc9H56y3NqWO33hIT4Y)

Hi there. Since i also have an ampera-e with a wican pro, did you ever get it working as you wished?

**Comment by Walleye0072** [comment]
I have a Chevy Bolt. I am using WiCAN-OBD with V4.14 firmware. 

Car Scanner runs, and I have a log from it with all the info that I want. AC Current, AC Voltage, SOC etc. 

Changed interface to AutoPID.  Clicked the Automate->"Scan PIDs" button. With the car ON, I see about 10 PIDs. Sometimes less. Sometimes more. 

Enabled MQTT and installed MQTT Explorer. I can see the topics for Standard PIDs only. I downloaded the BOLT config, and saved it. Restarted etc. No dice. I tried to add custom PIDs, similarly no dice. 

With HACS, I installed the integration on my supervised Home Automation System.  [https://github.com/jay-oswald/ha-wican](https://github.com/jay-oswald/ha-wican)

The integration is reading and publishing all the fields that I have setup: Standard PIDs, Vehicle Specific PIDs, Custom PIDs. I have attached a log file snippet. 
[WiCan HA Log.txt](https://github.com/user-attachments/files/20890774/WiCan.HA.Log.txt)

**Issues:**
1) I cannot get the Vehicle Specific nor Custom PIDs to populate data in neither Home Assistant nor MQTT Explorer. Standard PIDs work. 
2) I am getting different PIDs showing up when I click the Scan PIDs button. The ones I really need don't show up i.e. AC Current. 
3) If I knew the names I could use to add in the standard PIDs, could I do that in the config file and reload the WiCAN? 

**Bolt Car Scanner Log:** 
[WiCANODB2_logs.txt](https://github.com/user-attachments/files/20890838/WiCANODB2_logs.txt)

**WiCAN Sanitized Config File**
[config_2025-06-24 clean.json](https://github.com/user-attachments/files/20892440/config_2025-06-24.clean.json)
